backend/internal/storage/postgres/schema/user.go

```diff
@@ -178,14 +206,22 @@ func (r *UserRepository) UpdateUserOnboard(ctx context.Context, email string, en
 	return enthusiasm, nil
 }
 
-func (r *UserRepository) GetUserProfile(ctx context.Context, id uuid.UUID) (*models.Profile, error) {
+func (r *UserRepository) GetUserProfile(ctx context.Context, id uuid.UUID, idType string) (*models.Profile, error) {
 	profile := &models.Profile{}
-	query := `SELECT u.id, u.username, u.display_name, COUNT(DISTINCT followers.follower_id) AS follower_count, COUNT(DISTINCT followed.followee_id) AS followed_count
-		FROM "user" u
-		LEFT JOIN follower followers ON followers.followee_id = u.id
-		LEFT JOIN follower followed ON followed.follower_id = u.id
-		WHERE u.id = $1
-		GROUP BY u.id, u.username, u.display_name, u.profile_picture, u.bio;`
+	if !(idType == "id" || idType == "username" || idType == "displayName") {
+		print("idType must be 'username', 'displayName' or 'id'")
+		return nil, nil
+	}
+	query := fmt.Sprintf(`
+	SELECT u.id, u.username, u.display_name, 
+		   COUNT(DISTINCT followers.follower_id) AS follower_count, 
+		   COUNT(DISTINCT followed.followee_id) AS followed_count
+	FROM "user" u
+	LEFT JOIN follower followers ON followers.followee_id = u.id
+	LEFT JOIN follower followed ON followed.follower_id = u.id
+	WHERE u.%s = $1
+	GROUP BY u.id, u.username, u.display_name, u.profile_picture, u.bio;
+`, idType)
 
 	exists, err := r.UserExists(ctx, id.String())
```

Comment:
why have this query? seems unnecessary if your original query will fail when the user does not exist. seems to be an unnecessary round trip to the DB